File: src/datastructures/stack/Stack.py
class Node:
    def __init__(self, value):
        self.value: int = value
        self.next: Node = None


class Stack:
    def __init__(self, value):
        newNode: Node = Node(value)
        self.top: Node = newNode
        self.height: int = 1

    def printStack(self) -> None:
        temp: Node = self.top
        while temp != None:
            print(temp.value)
            temp = temp.next

    def push(self, value) -> None:
        newNode: Node = Node(value)
        if self.height == 0:
            self.top = newNode
        else:
            newNode.next = self.top
            self.top = newNode
        self.height += 1

    def pop(self) -> Node:
        if self.height == 0:
            return None
        temp: Node = self.top
        if self.height == 1:
            self.top.next = None
        else:
            self.top = self.top.next
            temp.next = None
        self.height -= 1
        return temp


myStack = Stack(1)
myStack.push(2)
print("pop value:", myStack.pop().value)
myStack.printStack()

File: src/datastructures/stack/Stack.java
package datastructures.stack;

public class Stack {
    private Node top;
    private int height;

    private class Node {
        private int value;
        private Node next;

        private Node(int value) {
            this.value = value;
        }
    }

    public Stack(int value) {
        Node newNode = new Node(value);
        top = newNode;
        height = 1;
    }

    public void printStack() {
        Node temp = top;
        while (temp != null) {
            System.out.println(temp.value);
            temp = temp.next;
        }
    }

    public void push(int value) {
        Node newNode = new Node(value);
        if (height == 0) {
            top = newNode;
        } else {
            newNode.next = top;
            top = newNode;
        }
        height++;
    }

    public Node pop() {
        if (height == 0) {
            return null;
        }
        Node temp = top;
        if (height == 1) {
            top = null;
        } else {
            top = top.next;
            temp.next = null;
        }
        height--;
        return temp;
    }
}
